runtime/ts/description.ts

```diff
@@ -176,15 +198,19 @@ export class DescriptionFormatter {
     }));
     const jointDescription = this._joinDescriptions(suggestions);
     if (jointDescription) {
-      if ((options || {}).skipFormatting) {
+      if (options.skipFormatting) {
         return jointDescription;
       } else {
         return this._capitalizeAndPunctuate(jointDescription);
       }
     }
+    return undefined;
   }
 
-  _joinDescriptions(strings) {
+  // TODO(mmandlis): the override of this function in subclasses also overrides the output. We'll need to unify
```

Comment:
hmm, interesting - this one returns a string, but the dom formatter returns an object... i guess we need a DescriptionOutput class then?

Reply:
There are other alternatives too:
(1) list the possible outputs as a union (but this means you need to know the possible types in the superclass)
(2) separate the subclass functionality out as not using the same method names.